client/wfnews-war/src/main/angular/src/app/components/base/base.component.ts

```diff
@@ -0,0 +1,292 @@
+    SAVE_SUCCESS_TEXT = "Saved Successfully";
+
+
+    DATE_FORMATS = DATE_FORMATS;
+
+
+    unsavedChangesMessage = "Unsaved Changes";
+
+    isUnsaved = false;
+    getElementInnerText = getElementInnerText;
+    isElementTruncated = isElementTruncated;
+
+    config: PaginationInstance = {
+        id: "Paginator",
+        itemsPerPage: 5,
+        currentPage: 1,
+        totalItems: 0
+    };
+
+    public paginatorLabels: any = {
+        previousLabel: "",
+        nextLabel: "",
+    };
+
+    constructor(protected router: Router,
+                protected route: ActivatedRoute,
+                protected sanitizer: DomSanitizer,
+                protected store: Store<RootState>,
+                protected fb: FormBuilder,
+                protected dialog: MatDialog,
+                protected applicationStateService: ApplicationStateService,
+                protected tokenService: TokenService,
+                protected connectionService: ConnectionService,
+                protected snackbarService: MatSnackBar,
+                protected overlay: Overlay,
+                protected cdr: ChangeDetectorRef,
+                protected appConfigService: AppConfigService,
+                protected http: HttpClient) {
+        this.initModels();
+        this.initializeConnectionService();
+    }
+
+    initComponentPreferences() {
+
+    }
+
+    initializeConnectionService() {
+        this.connectionService.monitor().subscribe(isConnected => {
+            this.isConnected = isConnected.hasNetworkConnection;
+        });
+    }
+
+    getPagingConfig(): PaginationInstance {
+        return this.config;
+    }
+
+
+    initModels() {
+
+    }
+
+    loadPage() {
+
+    }
+
+    save(etag?: string) {
+        this.isLocalSaving = true;
+    }
+
+    reload() {
+    }
+
+    saveOverwrite(etag: string) {
+        this.save(etag);
+    }
+
+    ngOnInit() {
+        this.loadPage();
+    }
+
+    ngAfterViewInit() {
+    }
+
+    ngOnChanges(changes: SimpleChanges) {
+        //console.log(changes);
+        if (changes.loadState) {
+            this.isLoading = changes.loadState.currentValue.isLoading;
+            let previousValue = changes.loadState.previousValue;
+            if (!this.isLoading && previousValue && previousValue.isLoading && !this.isLocalSaving) {
+                this.updateView();
+                this.invokeAfterLoaded();
+            }
+        }
+        if (changes.severeErrorState) {
+            this.isLocalSaving = false;
+        }
+        if (changes.errorState) {
+            this.errorState = changes.errorState.currentValue as ErrorState[];
+            if (this.viewModel.formGroup) {
+                this.viewModel.setErrorState(this.errorState);
+            }
+            if (this.isLocalSaving) {
+                if (this.errorState && this.errorState.length > 0) {
+                    this.isLocalSaving = false;
+                    // if (this.SAVE_FAIL_TEXT && !this.showingErrorDialog) {
```

Comment:
If this is not needed, remove it from the code.
Do we need to isLocalSaving checks?